### PR TITLE
Remove deprecated API

### PR DESCRIPTION
**Pull request by 1chandu**
- Remove deprecated API

Verified with `test-browser` and `examples`.

### CODE REVIEW

File: docs/upgrade-guide.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Upgrade Guide
 
-## Upgrading from deck.gl v4.1 to v4.2
+## Upgrading from deck.gl v4.1 to v5
```
Comment (ibgreen) [inline docs/upgrade-guide.md]: +1 Did you do a global search for 4.2?

File: docs/upgrade-guide.md [hunk 2]
```diff
@@ -52,6 +52,10 @@ All our layers enable depth test so we are going set this state during initializ
 For any custom needs, these parameters can be overwritten by updating them in [`onWebGLInitialized`](docs/api-reference/react/deckgl.md#onWebGLInitialized) callback or by passing them in `parameters` object to `drawLayer` method of `Layer` class.
 
 
+### assembleShaders
+
+`assembleShaders` was deprecated in v4.1 and now removed in v5. As described in v4.1 upgrade guide please use `Model` class instead.
```
Comment (ibgreen) [inline docs/upgrade-guide.md]: I would rephrase this:

The `assembleShaders` function was moved to luma.gl in v4.1 and is no longer re-exported from deck.gl. As described ...., or import it from luma.gl where it is defined.x

File: docs/whats-new.md [hunk 3]
```diff
@@ -78,7 +72,7 @@ Note: This change is mainly relevant to developers who write custom deck.gl laye
 
 ## Experimental Features
 
-As usual, deck.gl 4.2 contains a number of experimental features, e.g. multi viewport and first person viewport support. These features are still being finalized and the APIs have not been frozen, but can still be accessed by early adopters. See the roadmap article for more information on these.
+As usual, deck.gl 4.2 contains a number of experimental features, e.g. `multi viewport`, `first person viewport` and `viewport transitions`. These features are still being finalized and the APIs have not been frozen, but can still be accessed by early adopters. See the roadmap article for more information on these.
```
Comment (ibgreen) [inline docs/whats-new.md]: Nit: Back ticks are usually for code symbols like actual class and function names, maybe normal quotes are more appropriate?

File: docs/upgrade-guide.md (hunk 1 above)
Comment (1chandu) [reply docs/upgrade-guide.md]: Replaced in other places too.
